substdio.h

```diff
@@ -1,22 +1,36 @@
 #ifndef SUBSTDIO_H
 #define SUBSTDIO_H
 
+#include <sys/types.h>
+#include <unistd.h>
+
 typedef struct substdio {
   char *x;
   int p;
   int n;
   int fd;
-  int (*op)();
+  ssize_t (*op)();
 } substdio;
 
-#define SUBSTDIO_FDBUF(op,fd,buf,len) { (buf), 0, (len), (fd), (op) }
+#undef _QMAIL_HAS_C99
+#ifdef __STDC_VERSION__
+#if __STDC_VERSION__ >= 199901L
+#define _QMAIL_HAS_C99
+#endif
+#endif
+
+#ifdef _QMAIL_HAS_C99
+#define SUBSTDIO_FDBUF(o,f,buf,len) { .x = (buf), .p = 0, .n = (len), .fd = (f), .op = (o) }
+#else
+#define SUBSTDIO_FDBUF(o,f,buf,len) { (buf), 0, (len), (f), (o) }
+#endif
```

Comment:
Do we still need this? The C99 initializer is nicer and safer, of course.

Reply:
"Need" is a strong word. We can do it. I can also just rip it off, it will make no functional difference.

Reply:
I like the idea, for sure. My vote is it probably makes more sense once we’re ready to mandate C99, as a PR that does this unconditionally for all struct initializers.